librosa/core/spectrum.py

```diff
@@ -519,6 +520,450 @@ def ifgram(y, sr=22050, n_fft=2048, hop_length=None, win_length=None,
+    by the time offset from the window center.
+
+    See `reassigned_spectrogram` for references.
+
+    Parameters
+    ----------
+    y : np.ndarray [shape=(n,)], real-valued
+        audio time series
+
+    sr : number > 0 [scalar]
+        sampling rate of `y`
+
+    n_fft : int > 0 [scalar]
+        FFT window size
+
+    hop_length : int > 0 [scalar]
+        hop length, number samples between subsequent frames.
+        If not supplied, defaults to `win_length / 4`.
+
+    win_length : int > 0, <= n_fft
+        Window length. Defaults to `n_fft`.
+        See `stft` for details.
+
+    window : string, tuple, number, function, or np.ndarray [shape=(n_fft,)]
+        - a window specification (string, tuple, number);
+          see `scipy.signal.get_window`
+        - a window function, such as `scipy.signal.hanning`
+        - a user-specified window vector of length `n_fft`
+
+        See `stft` for details.
+
+        .. see also:: `filters.get_window`
+
+    dtype : numeric type
+        Complex numeric type for `S`. Default is 64-bit complex.
+
+    Returns
+    -------
+    times : np.ndarray [shape=(1 + n_fft/2, t), dtype=real]
+        Reassigned times:
+        `times[f, t]` is the time for bin `f`, frame `t`
+
+    S : np.ndarray [shape=(1 + n_fft/2, t), dtype=complex]
+        Short-time Fourier transform
+
+    Warns
+    --------
+    RuntimeWarning
+        Time estimates with zero support will produce a divide-by-zero warning
+        and will be returned as `np.nan`.
+
+    See Also
+    --------
+    stft : Short-time Fourier Transform
+    reassign_frequencies : Estimate frequency reassignments
+    reassigned_spectrogram : Time-frequency reassigned spectrogram
+
+    Examples
+    --------
+    >>> y, sr = librosa.load(librosa.util.example_audio_file())
+    >>> times, S = librosa.reassign_times(y, sr=sr)
+    >>> times
+    array([[ 0.077,  0.079,  ..., 61.362, 61.388],
+           [ 0.078,  0.077,  ..., 61.366, 61.538],
+           [ 0.088,  0.08 ,  ..., 61.358, 61.399],
+           ...,
+           [ 0.078,  0.077,  ..., 61.378, 61.372],
+           [ 0.082,  0.077,  ..., 61.371, 61.38 ],
+           [ 0.075,  0.076,  ..., 61.374, 61.385]])
+
+    """
+
+    # retrieve window samples if needed so that the time-weighted window can be
+    # calculated
+    if win_length is None:
+        win_length = n_fft
+
+    window = get_window(window, win_length, fftbins=True)
+    window = util.pad_center(window, n_fft)
+
+    # retrieve hop length if needed so that the frame times can be calculated
+    # here
+    if hop_length is None:
+        hop_length = int(win_length // 4)
+
+    # center is always False because padding will duplicate impulse components
+    # at the edges, which invalidates reassignment for edge frames
+    S_h = stft(
+        y=y,
+        n_fft=n_fft,
+        hop_length=hop_length,
+        window=window,
+        center=False,
```

Comment:
Same comment as before; consistency is more important than absolute correctness in this case.

Reply:
Done in https://github.com/librosa/librosa/pull/926/commits/9450b92e021de9f34cb5044957d9501c61098f20